### PR TITLE
Update reference to dns sidecar (was dnsmasq-metrics); remove exec-healthz

### PR DESCRIPTION
**Pull request by bowei**
-The image path is wrong -- I am waiting for the CI to pass here before pushing to google_containers-

### CODE REVIEW

**Comment by bowei** [comment]
@MrHohn

File: cluster/addons/dns/kubedns-controller.yaml.base [hunk 1]
```diff
@@ -137,33 +137,13 @@ spec:
...
-        - --cmd=nslookup kubernetes.default.svc.__PILLAR__DNS__DOMAIN__ 127.0.0.1 >/dev/null
-        - --url=/healthz-dnsmasq
-        - --cmd=nslookup kubernetes.default.svc.__PILLAR__DNS__DOMAIN__ 127.0.0.1:10053 >/dev/null
-        - --url=/healthz-kubedns
-        - --port=8080
-        - --quiet
-        ports:
-        - containerPort: 8080
-          protocol: TCP
+            memory: 20Mi
```
Comment (MrHohn) [inline cluster/addons/dns/kubedns-controller.yaml.base]: Discussed offline. May want to add cpu request here.

**Comment by MrHohn** [comment]
And should also keep `kubernetes/cluster/juju/layers/kubernetes/templates/kubedns-controller.yaml` in sync.

**Comment by bowei** [comment]
@thockin

**Pull request merged**